mne/decoding/ssd.py

```diff
@@ -0,0 +1,316 @@
+            if not isinstance(val, (int, float)):
+                raise ValueError(
+                    'Frequencies must be numbers, got %s' % type(val))
+        # check freq bands
+        if (filt_params_noise['l_freq'] > filt_params_signal['l_freq'] or
+                filt_params_signal['h_freq'] > filt_params_noise['h_freq']):
+            raise ValueError('Wrongly specified frequency bands!\n'
+                             'The signal band-pass must be within the noise '
+                             'band-pass!')
+        ch_types = {channel_type(info, ii)
+                    for ii in range(info['nchan'])}
+        if len(ch_types) > 1:
+            raise ValueError('At this point SSD only supports fitting '
+                             'single channel types. Your info has %i types' %
+                             (len(ch_types)))
+        self.info = info
+        self.freqs_signal = (filt_params_signal['l_freq'],
+                             filt_params_signal['h_freq'])
+        self.freqs_noise = (filt_params_noise['l_freq'],
+                            filt_params_noise['h_freq'])
+        self.filt_params_signal = filt_params_signal
+        self.filt_params_noise = filt_params_noise
+        self.sort_by_spectral_ratio = sort_by_spectral_ratio
+        if n_fft is None:
+            self.n_fft = int(self.info['sfreq'])
+        else:
+            self.n_fft = int(n_fft)
+        self.picks_ = (Ellipsis if picks is None else picks)
+        self.return_filtered = return_filtered
+        self.estimator = estimator
+        self.n_components = n_components
+        self.rank = rank
+        self.cov_method_params = cov_method_params
+
+    def _check_X(self, X):
+        """Check input data."""
+        if not isinstance(X, np.ndarray):
+            raise ValueError('X should be of type ndarray (got %s).'
+                             % type(X))
+        if X.ndim < 2:
+            raise ValueError('X must have at least 2 dimensions.')
+        elif X.ndim > 3:
+            raise ValueError('X must have at most 3 dimensions.')
+
+        n_chan = 0
+        if X.ndim == 2:
+            n_chan = X.shape[0]
+        elif X.ndim == 3:
+            n_chan = X.shape[1]
+        if n_chan != self.info['nchan']:
+            raise ValueError('Info must match the input data.'
+                             'Found %i channels but expected %i.' %
+                             (n_chan, self.info['nchan']))
+
+    def fit(self, X, y=None):
+        """Estimate the SSD decomposition on raw or epoched data.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) | shape (n_epochs,
+                n_channels, n_times)
+            The input data from which to estimate the SSD. Either 2D array
+            obtained from continuous data or 3D array obtained from epoched
+            data.
+        y : None | array, shape (n_samples,)
+                    Used for scikit-learn compatibility.
```

Comment:
This is indented too far
```suggestion
            Used for scikit-learn compatibility.
```